app/src/pages/inside/testCaseLibraryPage/hooks/useTestCases.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2025 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useState, useEffect } from 'react';
+import { TestCase } from 'components/testCaseList/testCaseCard/testCaseCard';
+import { mockTestCases } from 'components/testCaseList';
+
+interface UseTestCasesResult {
+  testCases: TestCase[];
+  loading: boolean;
+  hasTestCases: boolean;
+  setTestCases: (testCases: TestCase[] | ((prev: TestCase[]) => TestCase[])) => void;
+}
+
+export const useTestCases = (): UseTestCasesResult => {
+  const [testCases, setTestCases] = useState<TestCase[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  // Simulate data loading
+  useEffect(() => {
+    const loadTestCases = async () => {
+      setLoading(true);
+      // Simulate API call delay
+      await new Promise((resolve) => setTimeout(resolve, 1000));
+
+      // For demo purposes, use mock data
+      // In real implementation, you would fetch from API
+      setTestCases(mockTestCases);
+      setLoading(false);
+    };
+
+    loadTestCases();
+  }, []);
+
+  const hasTestCases = testCases && testCases.length > 0;
+
+  return {
+    testCases,
+    loading,
+    hasTestCases,
+    setTestCases,
```

Comment:
Do not expose setTestCases. For optimistic updates expose already bound methods:
```
optimisticDeleteTestCase(id: string) {
        setTestCases((prev) => prev.filter((tc) => tc.id !== id));
}
optimisticDuplicateTestCase(id: string) {
}
```